### ISSUE TITLE
URL of saved password should match top level frame URL

### ISSUE THREAD

**Issue by luckyrat** [kee-org/KeeFox#413]
Currently login forms filled via frames within a page will have their entry stored with the frame's URL as the entry's primary URL. This should have no impact on match accuracy but it does mean that one-click logins can often result in unusual behaviour (e.g. missing the outer part of the page or some kind of auto-redirection which then invalidates the login attempt).

We should instead store the entry with the top frame URL and add the frame URL to the list of alternative URLs.

The only downside of this new approach is that matched logins which previously worked across many sites as part of a shared/social login iframe will now be either duplicated or potentially displayed to the user with the page title and URL of a seemingly unrelated site.

Might also have to depend on #412 in order to ensure we match the right logins to the right frame with the right priority (but it might just work since we'll have the alt url exact match).

Low priority for 1.5

**Comment by luckyrat** [comment]
Unless we've somehow made this worse compared to v1.4 I'm going to drop this from v1.5.

I'm not ruling out implementing it in future but as well as addressing the downside of user-visible discrepancies across multiple sites, I've since realised that this same discrepancy will essentially mean that each login on a different website will trigger a "save password" prompt. It's open to debate but my initial feeling is that the problem of having no one-click login ability for some sites is not as severe as causing some sites to require "never ask for a password for this site" to be set for every site that includes the iframe login page. (Especially since no one has ever mentioned that this is a problem for them!)
